### PR TITLE
Add: instructions on how to run ruby templates

### PR DESCRIPTION
**Pull request by BKSpurgeon**
### What is this PR?

- New contributors might not know how to easily run the debugging templates (it took me a not insignificant time to learn how to do it). This PR adds instructions on how this can be done. Hopefully it will be of some use: if not, feel free discard it according to your discretion.

### What's the benefit of this PR?

* Hopefully this will lower the barriers to entry to new comers to help debug tests, and that will hopefully ease the maintenance burden for maintainers.

### CODE REVIEW

**Comment by hmistry** [comment]
Hi Ben

Thank you for the PR. Before merging it, I'd like to hear more since it's new for you. 

Could you share a bit more of all the things you encountered that was not trivial in using the debugging templates? 

Was using a Gemfile or installing gems not trivial for you or did you encounter issues with it?

I would like to see how we can make it beginner friendly because it can be a very easy to isolate problems when debugging or a quick way to explore things.

**Comment by BKSpurgeon** [comment]
HI hmistry

> Could you share a bit more of all the things you encountered that was not trivial in using the debugging templates?

Consider this bug report: https://github.com/shrinerb/shrine/issues/473

What are the steps required to solve it?

#### Steps taken:

> Was using a Gemfile or installing gems not trivial for you

- Alas, i simply didn't know how! nor did i know what gems should be installed. to further compound the problem: it took some time for me to work out why pry navigation commands weren't working: it turns out that `pry-nav` is packaged in a separate gem to `pry`.
- Also, users may want to debug after making changes to the shrine code itself, or they might want to debug directly in the shrine code: here's how i solved that problem:

```ruby
# Gemfile
gem 'shrine', path: "/home/koshy/UbuntuHDD/Documents/RailsProjects/OpenSourceProjects/shrine"
    # notice how the path is a local file.
```
and then to add a byebug statement within the shrine core code itself - i still couldn't get the debugger to stop within shrine itself, so i'll have to work out why that isn't working.

#### Solutions

- I would be very interested to know whether i'm doing things the right/sensible way.

rgds

Ben

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -123,6 +123,47 @@ post = Post.create(image: Down.download("https://example.com/image-from-internet
...
+  source 'https://rubygems.org'
+  gem "rails"
+  gem 'shrine'
+  gem "image_processing"
+  gem "sqlite3"
+end
+```
+
+* Now run `ruby template_name.rb` and the script should run.
+
```
Comment (hmistry) [inline CONTRIBUTING.md:165]: Lastly let's add a third section:

----
### Debugging Shrine code

If you would like to debug or step through Shrine's code using the above template scripts, follow these steps:
* Download (or clone) Shrine's repository to your local machine
`git clone ...`
* Checkout the version of shrine you are using by running `git checkout <version_tag>`
* Modify the `Shrine` gem line in your `Gemfile` to point to the local copy of Shrine code
`gem "shrine", path: "/path/to/your/local/shrine/code"`
* If needed, add a debugger like `byebug` or `pry` to the `Gemfile`
* Run `bundle install`
* Add `require "bundler/setup"` to the top of the template script
* You can add debugger statement anywhere in your template file or Shrine code
* Run `ruby template_name.rb`.

**Comment by hmistry** [comment]
@BKSpurgeon Thank you for sharing the context because I was thinking something else. I know where you're coming from because I went through the same thing myself a year ago. This PR will be very helpful and explains why not many people are using the templates to create self contained examples or doing quick experimentations with Shrine.

I've made comments and also want to add a third section on debugging setup. Please make the changes and it should be ready for merging.

File: CONTRIBUTING.md (hunk 1 above)
Comment (BKSpurgeon) [reply CONTRIBUTING.md:182]: thx got it working with the 'bundler/setup' line.

**Comment by BKSpurgeon** [comment]
@hmistry have incorporated the changes you've requested (have added minor comments where necessary).

**Comment by BKSpurgeon** [comment]
> @BKSpurgeon Thank you for sharing the context because I was thinking something else.

Out of sheer curiosity, what was the other content?

**Comment by hmistry** [comment]
This looks good. Thanks for making the changes.

The other context was using the template not for debugging but creating a small self contained example be it for experimenting or testing out your code. In this case, I wasn't sure why one would not think of creating a `Gemfile` or using `gem install` to install necessary gems to run the script like you would for any ruby script. But once you mentioned debugging, then I remembered it wasn't straight forward for me either, especially if your `ruby template.rb` doesn't run in bundler's context and you actually need either `require bundler/setup`  or run `bundle exec ruby template.rb`.

**Pull request merged**